cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/service/http/BasicSystemHttpServiceContext.java

```diff
@@ -137,4 +148,12 @@ public void run(TxRunnable runnable) throws TransactionException {
       throw new IllegalArgumentException(String.format("Namespace '%s' does not exist", namespace), nfe);
     }
   }
+
+  @Override
+  public byte[] runTask(RunnableTaskRequest runnableTaskRequest) throws Exception {
+    String systemAppClassName = SystemAppTask.class.getCanonicalName();
+    RunnableTaskRequest taskRequest = RunnableTaskRequest.getBuilder(systemAppClassName)
+      .withDelegate(runnableTaskRequest).build();
```

Comment:
I don't understand why the delegate is exposed as part of the task request. Each task should be self-contained. If it delegates to another task, it is its own implementation detail.

Reply:
Removed the delegate.